### PR TITLE
add basic js code and useful jquery plugins

### PR DESCRIPTION
**Pull request by visha12l**

### CODE REVIEW

File: js_basic_codes/area_of_circle.html [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+function circle_area(r)
+{
+  var r;
+  var area = 22/7*r*r;
+  return area;
+}
+var answer=circle_area(2);
+document.write(answer);
+*/
```
Comment (rishabhtatia1) [inline js_basic_codes/area_of_circle.html]: Blank spaces

File: js_basic_codes/area_of_circle.html [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+</head>
+<body>
+<div class="container">
+</div>
+ <script type="text/javascript">
+/* without passing the parameters
+function circle_area(radius)
+{
+  var area=22/7*radius*radius;
+  return area;
```
Comment (rishabhtatia1) [inline js_basic_codes/area_of_circle.html]: Indentation is not proper, should be equal everywhere

File: js_basic_codes/area_of_circle.html [hunk 3]
```diff
@@ -0,0 +1,44 @@
+<!DOCTYPE html>
+<html>
+<head>
+  <title>area of circle</title>
+</head>
+<body>
+<div class="container">
```
Comment (rishabhtatia1) [inline js_basic_codes/area_of_circle.html]: Indentation required

File: js_basic_codes/area_of_circle.html [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+document.write(answer);
+*/
+//area of circle using object and methods
+function Circle (radius) {
+    this.radius = radius;
+    this.area = function () {
+        return Math.PI * this.radius * this.radius;
+    };
+    // define a perimeter method here
+    this.perimeter=function(radius){
```
Comment (rishabhtatia1) [inline js_basic_codes/area_of_circle.html]: Space required near curly braces

File: js_basic_codes/count_digits_in_number.html [hunk 5]
```diff
@@ -0,0 +1,24 @@
+<!DOCTYPE html>
+<html>
+<head>
+  <title>count digit in number</title>
+</head>
+<body>
+<div class="container">
```
Comment (rishabhtatia1) [inline js_basic_codes/count_digits_in_number.html]: Indentation

File: js_basic_codes/count_digits_in_number.html [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+</div>
+<script type="text/javascript">
+
+var count=0;
+var num =prompt("enter hte number");//Math.floor() is javascript function used to make number into pur integer
+while (num=Math.floor(num))//it will run till num is not zero
+{
+  num = num/10;
```
Comment (rishabhtatia1) [inline js_basic_codes/count_digits_in_number.html]: Space required near operators

File: js_basic_codes/count_digits_in_number.html [hunk 7]
```diff
@@ -0,0 +1,24 @@
...
+  count++;
+document.write("Number :: "+ num + " count :: "+ count +"<br>");//br tag is used for breaking the line in javascript
+}
+alert(" \nthe number has "+ count +" digits");//ypu can use \n with only alerts
```
Comment (rishabhtatia1) [inline js_basic_codes/count_digits_in_number.html]: Spelling mistake in comment

File: js_basic_codes/emptyObject.html [hunk 8]
```diff
@@ -0,0 +1,16 @@
...
+<html>
+<head>
+  <title>Object creation in JS</title>
+</head>
+<body>
+<div class="container">
+console.dir() it gives more detais thn console.log
+</div>
+ <script type="text/javascript">
+var x = new Object();/* or var x={}; this will also ccreate an empty object*/
```
Comment (rishabhtatia1) [inline js_basic_codes/emptyObject.html]: Spelling mistake and indentation

File: js_basic_codes/function_multiply_with_parameters.html [hunk 9]
```diff
@@ -0,0 +1,25 @@
...
+<body>
+<div class="container">
+
+
+</div>
+ <script type="text/javascript">
+
+//function with passing the parameters
+
+function multiply(a,b){
```
Comment (rishabhtatia1) [inline js_basic_codes/function_multiply_with_parameters.html]: Space

File: js_basic_codes/function_multiply_with_parameters.html [hunk 10]
```diff
@@ -0,0 +1,25 @@
...
+
```
Comment (rishabhtatia1) [inline js_basic_codes/function_multiply_with_parameters.html]: Blank lines

File: js_basic_codes/function_pizzaToppings.html [hunk 11]
```diff
@@ -0,0 +1,29 @@
...
+</head>
+<body>
+<div class="container">
+
+
+</div>
+ <script type="text/javascript">
+ function takeOrder()
+ {
+   var a =prompt('please enter the pizza name');
```
Comment (rishabhtatia1) [inline js_basic_codes/function_pizzaToppings.html]: Space

File: js_basic_codes/function_pizzaToppings.html [hunk 12]
```diff
@@ -0,0 +1,29 @@
+<!DOCTYPE html>
+<html>
+<head>
+  <title>pizzaToppings</title>
+</head>
+<body>
+<div class="container">
```
Comment (rishabhtatia1) [inline js_basic_codes/function_pizzaToppings.html]: Indentation

File: js_basic_codes/jsConstructorDetails.html [hunk 13]
```diff
@@ -0,0 +1,26 @@
...
+</head>
+<body>
+  <script>
+  function Rabbit(adjective) {
+    this.adjective = adjective;
+    this.describeMyself = function() {
+        document.write("I am a " + this.adjective + " rabbit<br>");
+    };
+}
+
```
Comment (rishabhtatia1) [inline js_basic_codes/jsConstructorDetails.html]: Spaces

File: js_basic_codes/jsConstructorDetails.html [hunk 14]
```diff
@@ -0,0 +1,26 @@
...
+  <title>jsConstructorDetails</title>
+</head>
+<body>
+  <script>
+  function Rabbit(adjective) {
+    this.adjective = adjective;
+    this.describeMyself = function() {
+        document.write("I am a " + this.adjective + " rabbit<br>");
+    };
+}
```
Comment (rishabhtatia1) [inline js_basic_codes/jsConstructorDetails.html]: indentation

File: js_basic_codes/jsConstructorDetails.html [hunk 15]
```diff
@@ -0,0 +1,26 @@
...
+
+// now we can easily make all of our rabbits
+var rabbit1=new Rabbit('fluffy');//we have created new objects using rabbit constructor
+var rabbit2= new Rabbit('happy');
+var rabbit3= new Rabbit('sleepy');
+
+var answer=rabbit1.describeMyself(); //call to the constructor
```
Comment (rishabhtatia1) [inline js_basic_codes/jsConstructorDetails.html]: Spaces

File: js_basic_codes/jsConstructorDetails.html [hunk 16]
```diff
@@ -0,0 +1,26 @@
...
+    this.describeMyself = function() {
+        document.write("I am a " + this.adjective + " rabbit<br>");
+    };
+}
+
+// now we can easily make all of our rabbits
+var rabbit1=new Rabbit('fluffy');//we have created new objects using rabbit constructor
+var rabbit2= new Rabbit('happy');
+var rabbit3= new Rabbit('sleepy');
+
```
Comment (rishabhtatia1) [inline js_basic_codes/jsConstructorDetails.html]: Blank lines

File: js_basic_codes/jsGetAttribute.html [hunk 17]
```diff
@@ -0,0 +1,16 @@
...
+<head>
+  <title>get attribute in JS</title>
+</head>
+<body>
+<div class="container">
+  getAttribute is used to select the class ,src ,id attributes to an element
+   <h1 id="theTitle" class="highlight summer">What's happening?</h1>
+</div>
+ <script type="text/javascript">
+var element=document.querySelector('h1');
```
Comment (rishabhtatia1) [inline js_basic_codes/jsGetAttribute.html]: Spacing

File: js_basic_codes/jsMethodusingThis.html [hunk 18]
```diff
@@ -0,0 +1,47 @@
...
+{
+  this.age=newAge;
+  this.hobby=newHobby;
+}//this function will update both things
+
+var vishal=new Object();
+
+vishal.age=20;
+vishal.hobby="football";
+vishal.updateInfo=updateInfo;//this is the method
```
Comment (rishabhtatia1) [inline js_basic_codes/jsMethodusingThis.html]: Use of this??

File: js_basic_codes/jsObject.html [hunk 19]
```diff
@@ -0,0 +1,33 @@
...
+//method 1)object using literal notation
+var cricket={
+  god:'sachin',//please note there is no semicolon
+  age:10,
+  array:['this is array stored in object']
+}
+document.write(cricket.god,cricket.age,cricket.array)//print the object value
+//method 2)object  using constructor
+  var me=new Object();//This tells JavaScript: "I want you to make me a new thing, and I want that thing to be an Object.
+  me.name='vishal';//arrayName.property=value
```
Comment (rishabhtatia1) [inline js_basic_codes/jsObject.html]: Comment might be wrong

File: js_basic_codes/jsObjectasParameter.html [hunk 20]
```diff
@@ -0,0 +1,38 @@
...
+
+  // We can make a function which takes persons as arguments
+  // This one computes the difference in ages between two people
+  var ageDifference = function(person1, person2) {
+      return person1.age - person2.age;
+  };
+
+  // Make a new function, olderAge, to return the age of
+  // the older of two people
+  var olderAge=function(person1,person2){
```
Comment (rishabhtatia1) [inline js_basic_codes/jsObjectasParameter.html]: Spacing

File: js_basic_codes/jsValidations/characterValidation.html [hunk 21]
```diff
@@ -0,0 +1,31 @@
...
+var i
+while(1){
+    i=prompt( "ENTER A positive numeric : ");
+    if(typeof(i)==='number' || i>0)
+    {
+      break;
+    }
+}
+*/
+var i
```
Comment (rishabhtatia1) [inline js_basic_codes/jsValidations/characterValidation.html]: semicolon missing

File: js_basic_codes/jsValidations/characterValidation.html [hunk 22]
```diff
@@ -0,0 +1,31 @@
...
+while(1){
+    i=prompt( "ENTER A positive numeric : ");
+    if(typeof(i)==='string' || )
```
Comment (rishabhtatia1) [inline js_basic_codes/jsValidations/characterValidation.html]: no use of OR

File: js_basic_codes/jsViewportSize.html [hunk 23]
```diff
@@ -0,0 +1,26 @@
...
+  <title>jsViewPortSize</title>
+</head>
+<body>
+          <script>
+          window.addEventListener("resize", displayViewportSize, false);
+
+  function displayViewportSize(e) {
+    var viewportWidth = document.documentElement.clientWidth;
+    var viewportHeight = document.documentElement.clientHeight;
+    document.write(viewportWidth + '<br>');
```
Comment (rishabhtatia1) [inline js_basic_codes/jsViewportSize.html]: No need of break, I guess

File: js_basic_codes/js_OOPS/js_accessPrivateVariables.html [hunk 24]
```diff
@@ -0,0 +1,26 @@
...
+</head>
+<body>
+<div class="container">
+</div>
+<script type="text/javascript">
+function Person(name,age)
+{
+this.name=name;//public
+this.age=age;//public
+var salary=30000//private
```
Comment (rishabhtatia1) [inline js_basic_codes/js_OOPS/js_accessPrivateVariables.html]: semicolon

File: js_basic_codes/js_array.html [hunk 25]
```diff
@@ -0,0 +1,17 @@
+<!DOCTYPE html>
+<html>
+<head>
+  <title>jsArray</title>
+</head>
+<body>
+<div class="container">
+</div>
+ <script type="text/javascript">
+var friends=['manoj','parth','rao','prasad'] //here no curly braces
```
Comment (rishabhtatia1) [inline js_basic_codes/js_array.html]: semicolon

File: js_basic_codes/js_forLoop.html [hunk 26]
```diff
@@ -0,0 +1,16 @@
...
+<html>
+<head>
+  <title>jsForLoop</title>
+</head>
+<body>
+<div class="container">
+</div>
+ <script type="text/javascript">
+var friends=['manoj','parth','rao','prasad'] //here no curly braces
+for (var i=0;i<friends.length;i++){
```
Comment (rishabhtatia1) [inline js_basic_codes/js_forLoop.html]: Spacing

File: js_basic_codes/jshasOwnProperty.html [hunk 27]
```diff
@@ -0,0 +1,26 @@
...
+<script>
+//hasOwnProperty checks whether the object have
+var person=new Object();
+
+person.name="vishal";
+
+document.write(person.hasOwnProperty('name'));//it will return true as person object have the property called 'name'
+//now we will use if else loop
+
+if(person.hasOwnProperty('age')) {
```
Comment (rishabhtatia1) [inline js_basic_codes/jshasOwnProperty.html]: Space

File: js_basic_codes/number_table.html [hunk 28]
```diff
@@ -0,0 +1,17 @@
...
+<html>
+<head>
+  <title>number_table</title>
+</head>
+<body>
+<div class="container">
+</div>
+<script type="text/javascript">
+var num=2;
+for(i=1;i<=10;i++)
```
Comment (rishabhtatia1) [inline js_basic_codes/number_table.html]: Spacing

File: js_basic_codes/javascriptNavigationBar.html [hunk 29]
```diff
@@ -0,0 +1,104 @@
...
+    text-align: center;
+    padding: 14px 16px;
+    text-decoration: none;
+}
+
+li a:hover, .dropdown:hover .dropbtn {
+    background-color: red;
+}
+
+li.dropdown {
```
Comment (rishabhtatia1) [inline js_basic_codes/javascriptNavigationBar.html]: li should not be there

File: js_basic_codes/javascriptNavigationBar.html [hunk 30]
```diff
@@ -0,0 +1,104 @@
+<!DOCTYPE html>
+<html>
+<head>
+<style>
+ul {
```
Comment (rishabhtatia1) [inline js_basic_codes/javascriptNavigationBar.html]: CSS in the same file

File: js_basic_codes/addbtnOnClick.html [hunk 31]
```diff
@@ -0,0 +1,25 @@
...
+<head>
+  <title>add btn on button click</title>
+</head>
+<body>
+<div class="container">
+<button >click to create the button</button><!--this will not create any button-->
+</div>
+ <script type="text/javascript">
+
+
```
Comment (Karanpharate) [inline js_basic_codes/addbtnOnClick.html]: multiple blank lines not allowed.

File: js_basic_codes/addbtnOnClick.html [hunk 32]
```diff
@@ -0,0 +1,25 @@
...
+var buttons=document.querySelector('button');//select all buttons
+buttons.addEventListener('click', createBtn);//when button is clicked function is called
+
+function createBtn()
+{
+  var para =document.createElement('button');//new element is created
```
Comment (Karanpharate) [inline js_basic_codes/addbtnOnClick.html]: there should be space after =.

File: js_basic_codes/alert_onclick.html [hunk 33]
```diff
@@ -0,0 +1,27 @@
+<!DOCTYPE html>
+<html>
+<head>
+  <title>add btn on button click</title>
+</head>
+<body>
+<div class="container">
```
Comment (Karanpharate) [inline js_basic_codes/alert_onclick.html]: space before and after = operator .

File: js_basic_codes/addbtnOnClick.html [hunk 34]
```diff
@@ -0,0 +1,25 @@
...
+  para.textContent='this is your new button'.toUpperCase();//new content
+  document.body.appendChild(para);//append new element on body
+}
+
+ </script>y
```
Comment (Karanpharate) [inline js_basic_codes/addbtnOnClick.html]: remove y .

File: js_basic_codes/area_of_circle.html (hunk 4 above)
Comment (Karanpharate) [inline js_basic_codes/area_of_circle.html]: space before function declaration parenthesis.

File: js_basic_codes/area_of_circle.html [hunk 35]
```diff
@@ -0,0 +1,44 @@
...
+        return this.radius*2*Math.PI;
```
Comment (Karanpharate) [inline js_basic_codes/area_of_circle.html]: there should be space before and after operators

File: js_basic_codes/area_of_circle.html [hunk 36]
```diff
@@ -0,0 +1,44 @@
...
+        };
+};
+
+
```
Comment (Karanpharate) [inline js_basic_codes/area_of_circle.html]: avoid multiple blank lines.

File: js_basic_codes/count_digits_in_number.html [hunk 37]
```diff
@@ -0,0 +1,24 @@
...
+  <title>count digit in number</title>
+</head>
+<body>
+<div class="container">
+</div>
+<script type="text/javascript">
+
+var count=0;
+var num =prompt("enter hte number");//Math.floor() is javascript function used to make number into pur integer
+while (num=Math.floor(num))//it will run till num is not zero
```
Comment (Karanpharate) [inline js_basic_codes/count_digits_in_number.html]: space required before and after operator

File: js_basic_codes/dragonGame.html [hunk 38]
```diff
@@ -0,0 +1,32 @@
...
+<head>
+  <title>dragonGame</title>
+</head>
+<body>
+<div class="container">
+</div>
+ <script type="text/javascript">
+ var slaying=true;
+ var youHit=Math.floor(Math.random()*2);
+ var damageThisRound=Math.floor(Math.random()*5+1);
```
Comment (Karanpharate) [inline js_basic_codes/dragonGame.html]: space before and after operator required.

File: js_basic_codes/dragonGame.html [hunk 39]
```diff
@@ -0,0 +1,32 @@
...
+ var totalDamage=0;
+ while(slaying){
```
Comment (Karanpharate) [inline js_basic_codes/dragonGame.html]: space after while required

File: js_basic_codes/dragonGame.html [hunk 40]
```diff
@@ -0,0 +1,32 @@
...
+     if(youHit){
+         document.write('you hit the dragon');
+         totalDamage+=damageThisRound;
+         if(totalDamage>=4)
```
Comment (Karanpharate) [inline js_basic_codes/dragonGame.html]: space after if and before paranthesis required.

File: js_basic_codes/dragonGame.html [hunk 41]
```diff
@@ -0,0 +1,32 @@
...
+             {
+                 document.write('player slew the dragon ');
+                 }
+                 else{
```
Comment (Karanpharate) [inline js_basic_codes/dragonGame.html]: keep else statement on the same line as their curly braces .
there should be space between else statement and paranthesis

File: js_basic_codes/emptyObject.html [hunk 42]
```diff
@@ -0,0 +1,16 @@
...
+console.log(x);
+
```
Comment (Karanpharate) [inline js_basic_codes/emptyObject.html]: blank line

File: js_basic_codes/function_multiply_with_parameters.html (hunk 9 above)
Comment (Karanpharate) [inline js_basic_codes/function_multiply_with_parameters.html]: space before parenthesis

File: js_basic_codes/function_multiply_without_parameters.html [hunk 43]
```diff
@@ -0,0 +1,27 @@
...
+</head>
+<body>
+<div class="container">
+
+
+</div>
+ <script type="text/javascript">
+//function without passing the parameters
+
+ function multiply(a,b){
```
Comment (Karanpharate) [inline js_basic_codes/function_multiply_without_parameters.html]: space before parenthesis

File: js_basic_codes/function_multiply_without_parameters.html [hunk 44]
```diff
@@ -0,0 +1,27 @@
...
+   var a=10;
+   var b=3;
+   var product=a*b;
```
Comment (Karanpharate) [inline js_basic_codes/function_multiply_without_parameters.html]: space before and after operators

File: js_basic_codes/function_multiply_without_parameters.html [hunk 45]
```diff
@@ -0,0 +1,27 @@
...
+   return product;
+ }
+
+ document.write(multiply());
+
```
Comment (Karanpharate) [inline js_basic_codes/function_multiply_without_parameters.html]: blank lines

File: js_basic_codes/function_pizzaToppings.html [hunk 46]
```diff
@@ -0,0 +1,29 @@
...
+
```
Comment (Karanpharate) [inline js_basic_codes/function_pizzaToppings.html]: blank lines

File: js_basic_codes/function_pizzaToppings.html [hunk 47]
```diff
@@ -0,0 +1,29 @@
...
+
+</div>
+ <script type="text/javascript">
+ function takeOrder()
+ {
```
Comment (Karanpharate) [inline js_basic_codes/function_pizzaToppings.html]: parenthesis should start from same line as function declaration

File: js_basic_codes/function_pizzaToppings.html [hunk 48]
```diff
@@ -0,0 +1,29 @@
...
+   var a =prompt('please enter the pizza name');
+   return a;
+
+ }
+
+function getprice(orderCount)
```
Comment (Karanpharate) [inline js_basic_codes/function_pizzaToppings.html]: semicolon

File: js_basic_codes/function_pizzaToppings_return.html [hunk 49]
```diff
@@ -0,0 +1,29 @@
...
+<head>
+  <title>pizzaToppings with return</title>
+</head>
+<body>
+<div class="container">
+</div>
+ <script type="text/javascript">
+ var pizzaOrdered;//1
+ var orderCount=0;//2
+ function takeOrder(topping,crustType){//4
```
Comment (Karanpharate) [inline js_basic_codes/function_pizzaToppings_return.html]: space expected after coma and before parenthesis

File: js_basic_codes/function_pizzaToppings_return.html [hunk 50]
```diff
@@ -0,0 +1,29 @@
...
+   orderCount =orderCount + 1;//5
+   document.write('Order: "'+ crustType+' "pizza topped with "'+ topping +'"<br>');//6
+ }
+ function finalPrice(){//8
```
Comment (Karanpharate) [inline js_basic_codes/function_pizzaToppings_return.html]: space

File: js_basic_codes/function_pizzaToppings_return.html [hunk 51]
```diff
@@ -0,0 +1,29 @@
...
+ return initialPrice(orderCount) + tax(); //9
+}
+   function tax(){//10
+   return initialPrice(orderCount) * 0.06;//11
+ }
+ function initialPrice(itemCount){//12
+   return itemCount * 7.5;//13
+ }
+ takeOrder("","thin crust");//3
```
Comment (Karanpharate) [inline js_basic_codes/function_pizzaToppings_return.html]: space expected after comma.

File: js_basic_codes/importantProgram.html [hunk 52]
```diff
@@ -0,0 +1,29 @@
...
+<script>
+function Person(name,age,sex) //defining the constructor
+{
+  this.name=name;
+  this.age=age;
+  this.sex=sex;
+//defining an empty array
+var family=new Array;
+//using constructor
+family[0]=new Person("raosaheb",50,"male");
```
Comment (Karanpharate) [inline js_basic_codes/importantProgram.html]: space after comma

File: js_basic_codes/importantProgram.html [hunk 53]
```diff
@@ -0,0 +1,29 @@
...
+family[1]=new Person("kalpana",40,"female");
+family[2]=new Person("vishal",22,"male");
+family[3]=new Person("priyanka",20,"female");
+family[4]=new Person("satish",18,"male");
+//using for loop to print the information
+for(var i=0;i<family.length;i++)
```
Comment (Karanpharate) [inline js_basic_codes/importantProgram.html]: space after semicolon and operators and add parenthesis from same line

File: js_basic_codes/javascriptNavigationBar.html [hunk 54]
```diff
@@ -0,0 +1,104 @@
...
+
+.dropdown-content a {
+    color: black;
+    padding: 12px 16px;
+    text-decoration: none;
+    display: block;
+    text-align: left;
+}
+
+.dropdown-content a:hover {background-color: #f1f1f1}
```
Comment (Karanpharate) [inline js_basic_codes/javascriptNavigationBar.html]: indentation

File: js_basic_codes/jsCustomConstructor.html [hunk 55]
```diff
@@ -0,0 +1,24 @@
...
+<body>
+  <script>
+  //Constructors are a way to make objects with the keyword new
+  //The most basic constructor is the Object constructor, which will make an object with no methods or properties.
+  function Person(name,age){ //this is the function for Person constructor
+    this.name=name;
+    this.age=age;
+  } //by using this we can directly create number of person list only by giving the values
+
+var vishal=new Person('vishal',22);//person is the new constructor created by us
```
Comment (Karanpharate) [inline js_basic_codes/jsCustomConstructor.html]: space after comma

File: js_basic_codes/jsObjectasParameter.html [hunk 56]
```diff
@@ -0,0 +1,38 @@
...
+          if(person1.age>person2.age)
```
Comment (Karanpharate) [inline js_basic_codes/jsObjectasParameter.html]: use parenthesis

File: js_basic_codes/jsValidations/positiveNumber.html [hunk 57]
```diff
@@ -0,0 +1,20 @@
...
+  <title>positive number</title>
+</head>
+<body>
+<div class="container">
+</div>
+<script type="text/javascript">
+var i=0;
+while(1) {
+    i=prompt( "ENTER A POSITIVE NUMBER: ");
+    if(i>0 ){
```
Comment (Karanpharate) [inline js_basic_codes/jsValidations/positiveNumber.html]: spacing

File: js_basic_codes/jsValidations/positiveNumber.html [hunk 58]
```diff
@@ -0,0 +1,20 @@
...
+<html>
+<head>
+  <title>positive number</title>
+</head>
+<body>
+<div class="container">
+</div>
+<script type="text/javascript">
+var i=0;
+while(1) {
```
Comment (Karanpharate) [inline js_basic_codes/jsValidations/positiveNumber.html]: space after while.

File: js_basic_codes/js_OOPS/js_accessPrivateVariables.html (hunk 24 above)
Comment (Karanpharate) [inline js_basic_codes/js_OOPS/js_accessPrivateVariables.html]: semicolon missing

File: js_basic_codes/js_OOPS/js_inheritance.html [hunk 59]
```diff
@@ -0,0 +1,34 @@
...
+    this.numLegs=2;
+    }
+    // set its prototype to be a new instance of Animal
+
+    Penguin.prototype = new Animal();//this will tell to inherit the properties
+var penguin=new Penguin('pintu');
+penguin.sayName(); //now sayName method of animal class will also work for penguin due to inheritance
+
+
+
```
Comment (Karanpharate) [inline js_basic_codes/js_OOPS/js_inheritance.html]: multiple blank lines

File: js_basic_codes/js_boolean_if_else.html [hunk 60]
```diff
@@ -0,0 +1,19 @@
+<!DOCTYPE html>
+<html>
+<head>
+  <title>js_boolean_if_else</title>
+</head>
+<body>
+  <script>
+  var playcricket=true;//you can assign it to false
+  if(playcricket)
```
Comment (Karanpharate) [inline js_basic_codes/js_boolean_if_else.html]: parenthesis should be on same line as if statement

File: js_basic_codes/js_conditionalOperator.html [hunk 61]
```diff
@@ -0,0 +1,27 @@
+<!DOCTYPE html>
+<html>
+<head>
+  <title>jsconditionalOperator</title>
```
Comment (Karanpharate) [inline js_basic_codes/js_conditionalOperator.html]: use proper camel casing

File: js_basic_codes/js_continueLoop.html [hunk 62]
```diff
@@ -0,0 +1,49 @@
...
+/*loop will run like this
+for i=0 j=0,1,2,4
+for i=1 j=0,1,2,4
+for i=2 j=0,1,2,4
+when j=3 loop will skip this conidition so 3 will not be printed
+
+
+*/
+var btn=document.querySelector('button');
+btn.On('click',function(){
```
Comment (Karanpharate) [inline js_basic_codes/js_continueLoop.html]: space

File: js_basic_codes/js_forLoop.html [hunk 63]
```diff
@@ -0,0 +1,16 @@
+<!DOCTYPE html>
+<html>
+<head>
+  <title>jsForLoop</title>
+</head>
+<body>
+<div class="container">
+</div>
+ <script type="text/javascript">
+var friends=['manoj','parth','rao','prasad'] //here no curly braces
```
Comment (Karanpharate) [inline js_basic_codes/js_forLoop.html]: indentation

File: js_basic_codes/js_if_else_program1.html [hunk 64]
```diff
@@ -0,0 +1,33 @@
+<!DOCTYPE html>
+<html>
+<head>
+  <title>if else program</title>
+</head>
+<body>
+<div class="container">
+</div>
+<script>
+for (var i=1;i<=20;i++) //loop will run between 1 to 20
```
Comment (Karanpharate) [inline js_basic_codes/js_if_else_program1.html]: spacing

File: js_basic_codes/js_if_else_program1.html [hunk 65]
```diff
@@ -0,0 +1,33 @@
...
+{
+
+     if(i%3===0 && i%5===0)  //if number is deivisible by both print FizzBuzz
```
Comment (Karanpharate) [inline js_basic_codes/js_if_else_program1.html]: spacing

File: js_basic_codes/leapYear.html [hunk 66]
```diff
@@ -0,0 +1,36 @@
+<!DOCTYPE html>
+<html>
+<head>
+  <title>check leap year</title>
+</head>
+<body>
+<div class="container">
+
+    <input id="userInput" type="text" placeholder="enter year to check"/>
+    <button >check</button>
```
Comment (Karanpharate) [inline js_basic_codes/leapYear.html]: space in button tag not required

File: js_basic_codes/number_table.html (hunk 28 above)
Comment (Karanpharate) [inline js_basic_codes/number_table.html]: spacing

File: js_basic_codes/switchCaseProgram1.html [hunk 67]
```diff
@@ -0,0 +1,29 @@
...
+<body>
+<div class="container">
+</div>
+<script type="text/javascript">
+var getReview = function (movie) {
+switch (movie){
+
+    case "Toy Story 2" :
+        return "Great story. Mean prospector.";
+        break;s
```
Comment (Karanpharate) [inline js_basic_codes/switchCaseProgram1.html]: remove s

File: js_basic_codes/importantProgram.html [hunk 68]
```diff
@@ -0,0 +1,29 @@
+<!DOCTYPE html>
+<html>
+<head>
+  <title>important program using array,objects,constructor</title>
+</head>
+<body>
+<script>
+function Person(name,age,sex) //defining the constructor
+{
```
Comment (Karanpharate) [inline js_basic_codes/importantProgram.html]: parenthesis not closed(no } bracket found)

File: js_basic_codes/jsObjectMethod.html [hunk 69]
```diff
@@ -0,0 +1,28 @@
...
+*/
+  // here is bob again, with his usual properties
+var bob = new Object();
+bob.name = "Bob Smith";
+bob.age = 30;
+// this time we have added a method, setAge
+bob.setAge = function (newAge){
+  bob.age = newAge;
+};
+// here we set bob's age to 40
```
Comment (rishabhtatia1) [inline js_basic_codes/jsObjectMethod.html]: It's set to 20

**Pull request merged**